src/helpers/index.ts

```diff
@@ -30,6 +33,26 @@ export const getAuthentication = (
     return `JWT ${jwtToken}`;
   }
 
+  if (config.authentication?.oauth1) {
+    const oauthUrl = `${config.host}/plugins/servlet/oauth/`;
+    const oauth = new OAuth(
+      `${oauthUrl}request-token`,
+      `${oauthUrl}access-token`,
+      config.authentication.oauth1.consumerKey,
+      config.authentication.oauth1.consumerSecret,
+      '1.0',
+      null,
+      'RSA-SHA1',
+    );
+
+    return oauth.authHeader(
+      buildURL(buildFullPath(config.host, request.url), request.params),
```

Comment:
@MrRefactoring Proposal using axios, not yet fully tested. What do you think?

Reply:
@jayree I think this is a valid option, but I need to test it further

Reply:
@jayree I checked the code and the available axios API. Can we use the code below to implement the functionality? It looks more transparent and more susceptible to unforeseen changes by axios

```ts
    return oauth.authHeader(
      axios.getUri({ ...request, url: new URL(request.url!, config.host).toString() }),
      config.authentication.oauth1.accessToken,
      config.authentication.oauth1.tokenSecret,
      request.method!,
    );
```